server/k8s.go

```diff
@@ -239,12 +240,13 @@ func (w *k8sWatcherImpl) buildDetails(service *core.Service, externalServiceName
 	rs := &routableService{
 		externalServiceName: externalServiceName,
 		containerEndpoint:   net.JoinHostPort(clusterIp, port),
-		autoScaleUp:         w.buildScaleUpFunction(service),
+		autoScaleUp:         w.buildScaleFunction(service, 0, 1),
+		autoScaleDown:       w.buildScaleFunction(service, 1, 0),
 	}
 	return rs
 }
 
-func (w *k8sWatcherImpl) buildScaleUpFunction(service *core.Service) func(ctx context.Context) error {
+func (w *k8sWatcherImpl) buildScaleFunction(service *core.Service, from int, to int) ScalerFunc {
```

Comment:
nit: since you're needing to cast them to int32 for kubernetes API, perhaps just declare as int32 in the method args.